google-cloud-spanner/src/test/java/com/google/cloud/spanner/InstanceAdminClientImplTest.java

```diff
@@ -408,14 +531,53 @@ public void testUpdateInstanceWithNodeCountAndProcessingUnits() throws Exception
     when(rpc.updateInstance(
             instance,
             FieldMask.newBuilder()
-                .addAllPaths(Arrays.asList("display_name", "node_count", "labels"))
+                .addAllPaths(
+                    Arrays.asList("display_name", "autoscaling_config", "node_count", "labels"))
+                .build()))
+        .thenReturn(rawOperationFuture);
+    InstanceInfo instanceInfo =
+        InstanceInfo.newBuilder(InstanceId.of(INSTANCE_NAME))
+            .setInstanceConfigId(InstanceConfigId.of(CONFIG_NAME))
+            .setNodeCount(3)
+            .setProcessingUnits(3000)
+            .build();
+    OperationFuture<Instance, UpdateInstanceMetadata> operationWithFieldMask =
+        client.updateInstance(instanceInfo);
+    assertTrue(operationWithFieldMask.isDone());
+    assertEquals(INSTANCE_NAME, operationWithFieldMask.get().getId().getName());
+  }
+
+  @Test
+  public void testUpdateInstanceWithNodeCountAndProcessingUnitsAndAutoscalingConfig()
+      throws Exception {
+    com.google.spanner.admin.instance.v1.Instance instance =
+        com.google.spanner.admin.instance.v1.Instance.newBuilder()
+            .setName(INSTANCE_NAME)
+            .setConfig(CONFIG_NAME)
+            .setNodeCount(3)
+            .setProcessingUnits(3000)
+            .setAutoscalingConfig(getAutoscalingConfigProto())
+            .build();
+    OperationFuture<com.google.spanner.admin.instance.v1.Instance, UpdateInstanceMetadata>
+        rawOperationFuture =
+            OperationFutureUtil.immediateOperationFuture(
+                "updateInstance",
+                getAutoscalingInstanceProto(),
+                UpdateInstanceMetadata.getDefaultInstance());
+    // autoscaling_config should take precedence over node_count or processing_units when
+    // autoscaling_config is not null and no specific field mask is set by the caller.
+    when(rpc.updateInstance(
+            instance,
+            FieldMask.newBuilder()
+                .addAllPaths(Arrays.asList("display_name", "autoscaling_config", "labels"))
                 .build()))
         .thenReturn(rawOperationFuture);
     InstanceInfo instanceInfo =
         InstanceInfo.newBuilder(InstanceId.of(INSTANCE_NAME))
             .setInstanceConfigId(InstanceConfigId.of(CONFIG_NAME))
             .setNodeCount(3)
             .setProcessingUnits(3000)
+            .setAutoscalingConfig(getAutoscalingConfigProto())
             .build();
     OperationFuture<Instance, UpdateInstanceMetadata> operationWithFieldMask =
         client.updateInstance(instanceInfo);
```

Comment:
* Can we add more assertions on what we expect the node count, processing unit and autoscaling config to contain, similar to how we are asserting `INSTANCE_NAME` ?

Reply:
Same above.